src/dialog.jsx

```diff
@@ -145,13 +149,19 @@ let Dialog = React.createClass({
   componentWillReceiveProps (nextProps, nextContext) {
     let newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
     this.setState({muiTheme: newMuiTheme});
+
+    if( nextProps.isOpen !== this.props.isOpen ) {
```

Comment:
I think that we should also respond to a change of the defaultIsOpen. This is not the case here